catalog/resourcemanagement/resourcemanagement-usage-ui/src/main/java/org/codice/ddf/resourcemanagement/usage/service/DataUsage.java

```diff
@@ -0,0 +1,260 @@
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.resourcemanagement.usage.service;
+
+import java.lang.management.ManagementFactory;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import javax.management.InstanceAlreadyExistsException;
+import javax.management.InstanceNotFoundException;
+import javax.management.MBeanRegistrationException;
+import javax.management.MBeanServer;
+import javax.management.MalformedObjectNameException;
+import javax.management.NotCompliantMBeanException;
+import javax.management.ObjectName;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.LoggingLevel;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.commons.collections.MapUtils;
+import org.codice.ddf.persistence.PersistenceException;
+import org.codice.ddf.persistence.PersistentItem;
+import org.codice.ddf.persistence.PersistentStore;
+import org.codice.ddf.persistence.attributes.AttributesStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DataUsage extends RouteBuilder implements DataUsageMBean {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DataUsage.class);
+
+    // 23:30 Daily
+    public static final String DEFAULT_CRON_TIME = "0+30+23+*+*+?";
+
+    public static final String CRON_TIME_ID_KEY = "data-usage-cron-key";
+
+    public static final String CRON_TIME_KEY = "cron_time";
+
+    public static final String ID = "id";
+
+    public static final String TXT_PREFIX = "_txt";
+
+    private static final String PREFERENCES_TYPE = "preferences";
+
+    private static final String LNG_PREFIX = "_lng";
+
+    private AttributesStore attributesStore;
+
+    private PersistentStore persistentStore;
+
+    private ObjectName objectName;
+
+    private MBeanServer mBeanServer;
+
+    private String cronTime;
+
+    private ReadWriteLock readWriteLock = new ReentrantReadWriteLock();
+
+    public DataUsage() {
+        registerMbean();
+    }
+
+    public void init() {
+        this.cronTime = getPersistentCronTime();
+        LOGGER.debug("Set Cron Time as : {}", this.cronTime);
+    }
+
+    private String getPersistentCronTime() {
+        try {
+            List<Map<String, Object>> mapList;
+            try {
+                readWriteLock.readLock()
```

Comment:
`lock()` can block indefinitely, but can that condition happen with the context of how this code runs? If yes, then one of the tryLock() variants would be a good choice.